packages/core/test/merkle-distributor/MerkleDistributor.js

```diff
@@ -698,6 +688,31 @@ contract("MerkleDistributor.js", function(accounts) {
       });
     });
     describe("(claimMulti)", function() {
+      // Reorders the claimsArray array so that claims for the same
+      // account and same token are next to each other.
+      const sortClaimsByAccountAndToken = async claimsArray => {
+        // Since .sort() comparison function must be synchrous, preload
+        // all reward tokens into the claim data before sorting:
+        return (
+          await Promise.map(claimsArray, _claim => {
+            // Returns window data for each claim:
+            return merkleDistributor.merkleWindows(_claim.windowIndex);
+          })
+        )
+          .map((windowData, i) => {
+            // Gets the reward token for the window data for this claim
+            // and injects it into the claim object.
+            return {
+              ...claimsArray[i],
+              rewardToken: windowData.rewardToken
+            };
+          })
+          .sort((a, b) => {
+            // If a.account == b.account, then sorts by rewardToken
+            return a.account.localeCompare(b.account) || a.rewardToken.localeCompare(b.rewardToken);
```

Comment:
Took me a minute to digest this -- neat implementation of primary + secondary sort on strings where we don't care about order, just sort consistency and grouping.

Reply:
Yeah the previous implementation wasn't correctly sorting, because you can't assume that different window indices don't re-use reward tokens. (before I was sorting on window index + account).

Also, took me a while to realize that you can't make the `.sort()` comparator function `async` otherwise it just doesn't sort.